clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -890,12 +896,21 @@ public void unsubscribe() {
     public void assign(Collection<TopicPartition> partitions) {
         acquire();
         try {
-            log.debug("Subscribed to partition(s): {}", Utils.join(partitions, ", "));
-            this.subscriptions.assignFromUser(partitions);
-            Set<String> topics = new HashSet<>();
-            for (TopicPartition tp : partitions)
-                topics.add(tp.topic());
-            metadata.setTopics(topics);
+            if (partitions == null || partitions.isEmpty()) {
```

Comment:
For consistency with subscribe(), would it make sense to treat an empty collection as an implicit call to unsubscribe()? I realize the name was not chosen well (sorry!), but unsubscribe() is also intended to be used to reset an assignment.